File: OsmAnd/src/net/osmand/plus/helpers/enums/SpeedConstants.java
package net.osmand.plus.helpers.enums;

import android.content.Context;

import net.osmand.plus.R;

public enum SpeedConstants {
	KILOMETERS_PER_HOUR(R.string.km_h, R.string.si_kmh, false),
	MILES_PER_HOUR(R.string.mile_per_hour, R.string.si_mph, true),
	METERS_PER_SECOND(R.string.m_s, R.string.si_m_s, false),
	MINUTES_PER_MILE(R.string.min_mile, R.string.si_min_m, true),
	MINUTES_PER_KILOMETER(R.string.min_km, R.string.si_min_km, false),
	NAUTICALMILES_PER_HOUR(R.string.nm_h, R.string.si_nm_h, true);

	public final int key;
	public final int descr;
	public final boolean imperial;

	SpeedConstants(int key, int descr, boolean imperial) {
		this.key = key;
		this.descr = descr;
		this.imperial = imperial;
	}

	public String toHumanString(Context ctx) {
		return ctx.getString(descr);
	}

	public String toShortString(Context ctx) {
		return ctx.getString(key);
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/settings/enums/SpeedConstants.kt
package net.osmand.shared.settings.enums

import net.osmand.shared.util.Localization

enum class SpeedConstants(val key: String, val descr: String, val imperial: Boolean) {
	KILOMETERS_PER_HOUR("km_h", "si_kmh", false),
	MILES_PER_HOUR("mile_per_hour", "si_mph", true),
	METERS_PER_SECOND("m_s", "si_m_s", false),
	MINUTES_PER_MILE("min_mile", "si_min_m", true),
	MINUTES_PER_KILOMETER("min_km", "si_min_km", false),
	NAUTICALMILES_PER_HOUR("nm_h", "si_nm_h", true);

	fun toHumanString(): String {
		return Localization.getString(descr)
	}

	fun toShortString(): String {
		return Localization.getString(key)
	}
}